api/v1beta1/openstackmachine_types.go

```diff
@@ -33,6 +33,41 @@ const (
 	IPClaimMachineFinalizer = "openstackmachine.infrastructure.cluster.x-k8s.io/ip-claim"
 )
 
+// SchedulerHintAdditionalValue represents the value of a scheduler hint property.
+// The value can be of various types: bool, string, int, or an object.
+// The Type field indicates the type of the value being used.
+type SchedulerHintAdditionalValue struct {
+	// Type represents the type of the value (e.g., "bool", "string", "int").
+	Type string `json:"type"`
```

Comment:
Please can you define this as an enum, both in Go and in the CRD. Example: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/90da86dc16a85a6100a07ab3a5a9d70389ddb8bb/api/v1beta1/types.go#L546-L554